pkg/operator/osd/osd.go

```diff
@@ -159,13 +159,18 @@ func (c *Cluster) podTemplateSpec(devices []Device, directories []Directory, sel
 		dataDirSource = v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: c.dataDirHostPath}}
 	}
 
-	// create volume config for the data dir and /dev so the pod can access devices on the host
 	volumes := []v1.Volume{
 		{Name: k8sutil.DataDirVolume, VolumeSource: dataDirSource},
-		{Name: "devices", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/dev"}}},
 		k8sutil.ConfigOverrideVolume(),
 	}
 
+	// by default, don't define any volume config unless it is required
+	if len(devices) > 0 {
```

Comment:
in addition to the list of specific devices, there are also fields that influence whether or not devices should be used by the OSD pod.  the `selection` arg to this function also has fields like that, so they must be taken into account too.  It's fields can be found here:  https://github.com/rook/rook/blob/master/pkg/operator/osd/spec.go#L52

and here's an example of where these other possible "should devices be used" fields are getting checked: https://github.com/rook/rook/blob/master/pkg/operator/osd/osd.go#L224

instead of `if len(devices) > 0`, perhaps the check should be something like:
```
len(devices) > 0 || selection.DeviceFilter != "" || selection.getUseAllDevices() || selection.MetadataDevice != ""
```

if any of those are true, then it means devices should be used in the OSD pod.  perhaps do that check once at the top of the func, save it to a var, then just check that var in each of the subsequent if checks.